src/parser/Expression.ts

```diff
@@ -429,8 +431,8 @@ export class IndexedGetExpression extends Expression {
             ...this.obj.transpile(state),
             this.questionDotToken ? state.transpileToken(this.questionDotToken) : '',
             state.transpileToken(this.openingSquare),
-            ...this.index.transpile(state),
-            state.transpileToken(this.closingSquare)
+            ...(this.index ? this.index.transpile(state) : []),
```

Comment:
This can be slightly simplified:
```suggestion
            ...(this.index?.transpile(state) ?? []),
```